src/components/FilterBarPopup/FilterBarPopup.tsx

```diff
@@ -0,0 +1,195 @@
+  /** A number that represents the total number of results in the associated search results list. */
+  totalResults?: number;
+}
+
+export const FilterBarPopup: ChakraComponent<
+  React.ForwardRefExoticComponent<
+    React.PropsWithChildren<FilterBarPopupProps> &
+      React.RefAttributes<HTMLDivElement>
+  >,
+  FilterBarPopupProps
+> = chakra(
+  forwardRef<HTMLDivElement, React.PropsWithChildren<FilterBarPopupProps>>(
+    (props, ref?) => {
+      const {
+        children,
+        id,
+        isOpen,
+        heading,
+        onClear,
+        onSubmit,
+        onOpen,
+        onClose,
+        renderChildren,
+        selectedItems,
+        totalResults,
+        ...rest
+      } = props;
+
+      const styles = useMultiStyleConfig("FilterBarPopup", {
+        width: "full",
+      });
+
+      const generalHeadingProps = {
+        size: "heading5" as HeadingSizes,
+        noSpace: true,
+      };
+      // If `heading is a string, then we want the default heading,
+      // otherwise, use whatever the user passed in.
+      const finalHeading = heading ? (
+        typeof heading === "string" ? (
+          <Heading level="h2" text={heading} {...generalHeadingProps} />
+        ) : (
+          React.cloneElement(heading, generalHeadingProps)
+        )
+      ) : null;
+
+      // Chakra's hook to control Modal's actions.
+      const disclosure = useDisclosure();
+      const finalIsOpen = isOpen ? isOpen : disclosure.isOpen;
+      const finalOnOpen = onOpen ? onOpen : disclosure.onOpen;
+      const finalOnClose = onClose ? onClose : disclosure.onClose;
+
+      const onSubmitAndClose = () => {
+        onSubmit();
+        finalOnClose();
+      };
+
+      return (
+        <Box id={`filter-bar-${id}`} ref={ref} {...rest}>
+          <Button
+            id={`filter-bar-${id}-show-filters`}
+            buttonType="secondary"
+            onClick={finalOnOpen}
+          >
+            {`Show filters`}
+          </Button>
+          <Modal
+            isOpen={finalIsOpen}
+            onClose={finalOnClose}
+            size={{ base: "full", md: "md" }}
+            scrollBehavior="inside"
+          >
+            <ModalOverlay />
+            <ModalContent>
+              <ModalHeader sx={finalHeading ? styles.modalHeader : undefined}>
+                {finalHeading}
+              </ModalHeader>
+              <ModalCloseButton sx={styles.modalCloseButton} />
+              <ModalBody sx={styles.modalBody}>
+                <Stack direction="column" rowGap="s" spacing="0">
+                  {renderChildren({
+                    isBlockElement: true,
+                    layout: "column",
+                    width: "full",
+                  })}
+                </Stack>
+              </ModalBody>
+              <ModalFooter
+                sx={
+                  onSubmit || onClear || totalResults
+                    ? styles.modalFooter
+                    : undefined
+                }
```

Comment:
The conditional logic is not applicable in this component. The `Show results` button will always be visible in `FilterBarPopup`, so these styles should always be applied